platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/streams/SerializableDataInputStream.java

```diff
@@ -248,41 +312,39 @@ private <T extends SelfSerializable> void readSerializableIterableWithSizeIntern
         final ValueReference<Integer> version = new ValueReference<>();
 
         for (int i = 0; i < size; i++) {
-            final T next =
-                    readNextSerializableIteration(allSameClass, readClassId, classId, version, serializableConstructor);
+            final T next = readNextSerializableIteration(
+                    allSameClass, readClassId, classId, version, serializableConstructor, permissibleCLassIds);
             callback.accept(next);
         }
     }
 
     /**
-     * Helper method for {@link #readSerializableIterableWithSize(int, Consumer)}. Protected instead of
-     * private to allow debug framework to intercept this method.
+     * Helper method for {@link #readSerializableIterableWithSizeInternal(int, boolean, CheckedFunction, Consumer, Set)}
+     * . Protected instead of private to allow debug framework to intercept this method.
      *
-     * @param allSameClass
-     * 		true if the elements all have the same class
-     * @param readClassId
-     * 		if true then the class ID needs to be read, ignored if allSameClass is true
-     * @param classId
-     * 		the class ID if known, otherwise null
-     * @param version
-     * 		the version if known, otherwise ignored
-     * @param serializableConstructor
-     * 		given a class ID, returns a constructor for that class
-     * @param <T>
-     * 		the type of the elements in the sequence
+     * @param allSameClass            true if the elements all have the same class
+     * @param readClassId             if true then the class ID needs to be read, ignored if allSameClass is true
+     * @param classId                 the class ID if known, otherwise null
+     * @param version                 the version if known, otherwise ignored
+     * @param serializableConstructor given a class ID, returns a constructor for that class
+     * @param permissibleCLassIds     a set of class IDs that are allowed to be read, will throw an IOException if asked
+     *                                to deserialize a class not in this set, all class IDs are permitted if null.
+     *                                Ignored if readClassId is false.
+     * @param <T>                     the type of the elements in the sequence
      * @return true if the class ID has already been read
      */
     protected <T extends SelfSerializable> T readNextSerializableIteration(
             final boolean allSameClass,
             final boolean readClassId,
-            final ValueReference<Long> classId,
-            final ValueReference<Integer> version,
-            final CheckedFunction<Long, T, IOException> serializableConstructor)
+            @NonNull final ValueReference<Long> classId,
+            @NonNull final ValueReference<Integer> version,
+            @NonNull final CheckedFunction<Long, T, IOException> serializableConstructor,
+            @Nullable final Set<Long> permissibleCLassIds)
             throws IOException {
 
         if (!allSameClass) {
```

Comment:
The logic if `allSameClass` is `false` takes `permissibleClassIds` into account, but what about checking class id when `allSameClass` is `true`? I'm not seeing any consideration of class id in that case

Reply:
Good catch. Fixed.